Makefile

```diff
@@ -595,25 +595,9 @@ html-docs-dockerized: build-test ## Build html docs dockerized
 		$(MLRUN_TEST_IMAGE_NAME_TAGGED) \
 		make html-docs
 
-.PHONY: fmt
-fmt: ## Format the code using Ruff
-	@echo "Running ruff checks and fixes..."
-	python -m ruff check --fix-only
-	python -m ruff format
-
-.PHONY: lint-imports
-lint-imports: ## Validates import dependencies
-	@echo "Running import linter"
-	lint-imports
-
 .PHONY: lint
-lint: fmt-check lint-imports ## Run lint on the code
-
-.PHONY: fmt-check
-fmt-check: ## Check the code (using ruff)
-	@echo "Running ruff checks..."
-	python -m ruff check --exit-non-zero-on-fix
-	python -m ruff format --check
+lint:
+	pre-commit run --all-files
```

Comment:
Basically what this does is equivalent to lint then fmt. If the format is wrong, a subsequent run will pass because the first run already formatted the code.

That's probably fine, but I thought it useful to point out the change in behaviour.

Reply:
Right, good point. I'll write it in the message.
One may choose a Ruff integration for their IDE and catch this even earlier. I recommend Ruff's VSCode extension for VSCode users.